### PR TITLE
Refactor CLI classes

### PR DESCRIPTION
**Pull request by mjankowski**
Followup on https://github.com/mastodon/mastodon/pull/23978

This makes some changes to the CLI-related classes that are called by `tootctl` script...

- Moves all classes under a `CLI::` namespace, and renames file to be in a cli/ folder (instead of _cli.rb names)
- Misc rubocop shuffling to accomodate that movement
- Add a `CLI::Base` class which all subcommands inherit from, and which includes common methods
- Move some of the previous `CLIHelper` module methods into the base class
- Rename the `CLIHelper` module to `CLI::ProgressHelper` since all remaining methods are related to progress bar output
- I tried to chip away a bit at some of the rubocop todo items where it was easy to do so
- Overall spec coverage is improved a bit since #23978 which just got some files to load, but is not close to 100% yet. Some of the code paths here are tricky to exercise in specs or slightly awkward to work with.

In theory the `tootctl` command and all subcommand functionality should not have changed here.

Opening as draft for now to get initial feedback - and because I want to use the PR review process to leave a few of my own comments on things I had questions about and encountered while in here. There are also a few exceptions to the last statement (that nothing changed) and I will leave comments about those to call them out while still in draft.

Because of the bulk file move here this might be hard to review from just the diff and it may be easier to step through a commit at a time or something. Not sure. If that's a huge issue, I could open a separate PR with *just the move*, get that merged in first, then step through these other changes in subsequent PR where seeing the diff may be easier.

### CODE REVIEW

**Comment by nschonni** [comment]
Wonder if it should be a root `tootctl` folder, but I don't really know what the convention is for that

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -1551,13 +1546,7 @@ Rails/NegateInclude:
...
     - 'lib/paperclip/color_extractor.rb'
 
-# This cop supports unsafe autocorrection (--autocorrect-all).
-# Configuration parameters: Include.
-# Include: app/**/*.rb, config/**/*.rb, db/**/*.rb, lib/**/*.rb
-Rails/Output:
-  Exclude:
-    - 'lib/mastodon/ip_blocks_cli.rb'
-
+# Offense count: 9
```
Comment (nschonni) [inline .rubocop_todo.yml:1549]: If you run `bundle exec rubocop --regen-todo` it should generate it with the current settings that won't add these anymore
```suggestion
```

**Comment by mjankowski** [comment]
There's a failure here related to redis connections...

```
 Failure/Error: RedisConfiguration.pool.checkin if Thread.current[:redis]

     ConnectionPool::Error:
       no connections are checked out
```

I believe this is because the `reset_connection_pools!` method in the CLI classes (which are now actually being run in specs, after these changes) is leaving things in a state where there are order dependent failures. Need to investigate this more and figure out how to safely run that code while leaving things in a consistent state between specs.

**Comment by mjankowski** [comment]
Rebased all this against https://github.com/mastodon/mastodon/pull/24139 - will probably take same approach here as the rubocop clean up stuff ... try to lump all this into smaller one off PRs; then close this once done.

**Comment by mjankowski** [comment]
Closing in favor of the various things extracted from this. To the extent there's more in here, I'll repeat it if/when the others are mreged.

**Pull request closed without merging**